static/js/components/Navigation_test.js

```diff
@@ -8,52 +8,90 @@ import sinon from "sinon"
 import * as util from "../lib/util"
 
 describe("Navigation", () => {
-  let sandbox, userIsAnonymousStub
+  let sandbox, userIsAnonymousStub, defaultProps: Object, userCanPostStub
 
   beforeEach(() => {
     sandbox = sinon.sandbox.create()
     userIsAnonymousStub = sandbox.stub(util, "userIsAnonymous")
     userIsAnonymousStub.returns(false)
+    userCanPostStub = sandbox.stub(channels, "userCanPost")
+    userCanPostStub.returns(true)
+
+    const subscribedChannels = makeChannelList(10)
+    defaultProps = {
+      pathname:           "/",
+      subscribedChannels: subscribedChannels,
+      channels:           new Map(
+        subscribedChannels.map(channel => [channel.name, channel])
+      )
+    }
   })
 
   afterEach(() => {
     sandbox.restore()
   })
 
-  const defaultProps = { pathname: "/", subscribedChannels: [] }
   const renderComponent = (props = defaultProps) =>
     shallow(<Navigation {...props} />)
 
-  it("create post link should not have channel name if channelName is not in URL", () => {
-    const wrapper = renderComponent()
-    assert.lengthOf(wrapper.find(Link), 2)
-    const props = wrapper
-      .find(Link)
-      .at(0)
-      .props()
-    assert.equal(props.to, "/create_post/")
-    assert.equal(props.children, "Submit a New Post")
-  })
+  describe("create post link", () => {
+    it("should not have channel name if channelName is not in URL", () => {
+      const wrapper = renderComponent()
+      assert.lengthOf(wrapper.find(Link), 2)
+      const props = wrapper
+        .find(Link)
+        .at(0)
+        .props()
+      assert.equal(props.to, "/create_post/")
+      assert.equal(props.children, "Submit a New Post")
+    })
 
-  it("create post link should have channel name if channelName is in URL", () => {
-    const wrapper = renderComponent({
-      ...defaultProps,
-      pathname: "/channel/foobar"
+    it("create post link should have channel name if channelName is in URL", () => {
+      const wrapper = renderComponent({
+        ...defaultProps,
+        pathname: "/channel/foobar"
+      })
+      const link = wrapper.find(Link).first()
+      assert.equal(link.props().to, newPostURL("foobar"))
+      assert.equal(link.props().children, "Submit a New Post")
     })
-    const link = wrapper.find(Link).first()
-    assert.equal(link.props().to, newPostURL("foobar"))
-    assert.equal(link.props().children, "Submit a New Post")
-  })
 
-  it("should not show the create post link if an anonymous user", () => {
-    userIsAnonymousStub.returns(true)
-    const wrapper = renderComponent()
-    assert.isNotOk(wrapper.find(".mdc-button").exists())
+    it("should not show the create post link if an anonymous user", () => {
+      userIsAnonymousStub.returns(true)
+      const wrapper = renderComponent()
+      assert.isNotOk(wrapper.find(".mdc-button").exists())
+    })
+
+    it("should not show the create post link if the channel can't be shown to the user", () => {
```

Comment:
should this test (and the one below) read `"if the user can't post in the channel", instead of being about showing the channel to the user?